farm/data_handler/input_features.py

```diff
@@ -410,6 +410,77 @@ def sample_to_features_qa(sample, tokenizer, max_seq_len, sp_toks_start, sp_toks
                     "seq_2_start_t": seq_2_start_t}
     return [feature_dict]
 
+def sample_to_features_dpr(
+    sample, max_seq_len, query_tokenizer, passage_tokenizer, embed_title=True, remove_sep_tok_from_untitled_passages=True
+):
+    """
+    Generates a dictionary of features for a given input sample that is to be consumed by a dense passage retrieval model.
+
+    :param sample: Sample object that contains human readable text and label fields from a single text classification data sample
+    :type sample: Sample
+    :param max_seq_len: Sequences are truncated after this many tokens
+    :type max_seq_len: int
+    :param query_tokenizer: A tokenizer object that can turn the query sentence into a list of tokens
+    :return: A list with one dictionary containing the keys "input_ids", "padding_mask" and "segment_ids". The values are lists containing those features.
+    :rtype: list
+    :param passage_tokenizer: A tokenizer object that can turn all context passages into a list of tokens
+    :return: A list with one dictionary containing the keys "input_ids", "padding_mask" and "segment_ids". The values are lists containing those features.
+    :rtype: list
+    :embed_title: whether to encode the context passage titles with the context passages
+    :type embed_title: bool
+    """
+
+    tokens_query = sample.tokenized["query_tokens"]
+    tokens_positive_ctx = sample.tokenized["positive_ctx_tokens"]
+    tokens_positive_ctx_title = sample.tokenized.get("positive_ctx_title_tokens", None)
+    tokens_hard_negative_ctx = sample.tokenized["hard_negative_ctx_tokens"]
+    tokens_hard_negative_ctx_title = sample.tokenized.get("hard_negative_ctx_titles_tokens", None)
+
+    query_inputs = query_tokenizer.encode_plus(
+        text=tokens_query["tokens"],
+        max_length=max_seq_len,
+        add_special_tokens=True,
+        truncation_strategy='do_not_truncate',
+        padding="max_length",
+        return_token_type_ids=True,
+        is_pretokenized=True
```

Comment:
1) We probably want to mostly use fast tokenizers here  
2) is_pretokenized works differently than we thought initially. It expects "words" but not "subword tokens". We'll rather need to pass in the "raw text" here. 
=> Maybe we should only support Fast Tokenizers in DPR as it will almost always be a heavy task on big datasets?
=> Then we could also do tokenization once (via encode_plus / batch_encode_plus) and get rid of FARM's tokenize_with_metadata